### ISSUE TITLE
Rule Proposal: `vue/no-unused-vars`

### ISSUE THREAD

**Issue by mysticatea** [vuejs/eslint-plugin-vue#100]
## Please describe what the rule should do:

`no-unused-vars` should report variable definitions of `v-for` directives or `scope` attributes if those are not used.

## What category of rule is this? (place an "X" next to just one item)

[ ] Enforces code style
[X] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

## Provide 2-3 code examples that this rule will warn about:

```html
<template>
    <ol v-for="i in 5"><!-- "i" is defined but never used. -->
        <li>item</li>
    </ol>
</template>
<script>
export default {
}
</script>
```

**Comment by michalsnik** [comment]
Okay, this one seems perfectly doable and makes perfect sense :)

**Comment by michalsnik** [comment]
Though maybe we should call it `no-unused-template-vars` ?

**Comment by sylvainpolletvillard** [comment]
@mysticatea  @michalsnik 
I don't understand the purpose of this rule. There are legit usecases for the code example provided. In my Vue training course, I have this:
```vue
<span class="rating>
   <template v-for="n in nbStars">★</template>
</span>
````

to show a certain amount of stars based on a rating. I don't care about the `n` variable here. Why would this be considered an error ? Please reconsider this rule.
